sklearn/manifold/spectral_embedding_.py

```diff
@@ -276,7 +279,9 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
             raise ValueError
 
     elif eigen_solver == "lobpcg":
-        laplacian = laplacian.astype(np.float)  # lobpcg needs native floats
+        # lobpcg needs native floats
+        laplacian = check_array(laplacian, dtype=np.float, 
+                                accept_sparse=True, copy=True)
```

Comment:
the comment says lobpcg needs native float.... I'm not sure where that comes from, though...